updates for metadata and manuscript finalization

gregcaporaso opened the pull request.

gregcaporaso committed:
integrate legend in Visualization

if `export_legend` is enabled, present the legend in the resulting
Visualization.
Note that there was a minor bug in the original version of the
Visualizer - since the conditional was being evaluated after
`export_legend` was cast to a string, it always evaluated to `True`.

gregcaporaso committed:
updates to reflect refactored metadata

gregcaporaso commented:
@lizgehret, this should be updated for the new metadata. Here's the bucket 3/4 plot that you create in your example command - does this align with your previous version? 

Figure:
![image](https://github.com/user-attachments/assets/0c1c2898-cfbb-453e-9ea6-ccce87a6a688)

Legend:
![image](https://github.com/user-attachments/assets/698137de-9d00-4c79-8659-cc0a7f1fe74a)

lizgehret commented:
Hey @gregcaporaso, I think there's a little confusion around the `export_legend` functionality - it should only be included in the main plot if `export_legend` is set to False. The purpose of this is to allow for Jeff to combine multiple bucket figures into a faceted view in biorender, with the legend exported as a separate figure that he can add in where it stylistically makes sense. The expected behavior is as follows:

if `export_legend` == False:
Only the pcoa_plot.png should be included in the data directory, and it should include the legend in the main figure.
if `export_legend` == True:
The pcoa_plot.png and legend.png should both be included in the data directory, and the legend should not be included in the main figure.

I just tested this again and confirmed it's working as expected. Do you think it would be helpful to change the parameter name to prevent any confusion on what the expected behavior is?

lizgehret commented:
Re: the figure looking the same - it looks very similar, but not exactly identical. Here's what my version looks like:
![pcoa_plot](https://github.com/user-attachments/assets/8b4be425-6bf0-48ae-961e-2b08acb58280)

I'm just seeing a few differences in the bulking material samples, which we may want to look at together during the composting meeting (just to make sure none of the Metadata changes unintentionally changed the identification of any particular samples).

lizgehret commented on gut_to_soil_manuscript_figures/_methods.py:
```diff
@@ -44,12 +44,11 @@ def pcoa_2d(output_dir: str, metadata: qiime2.Metadata,
     himalaya = str(himalaya)
     pit_toilet = str(pit_toilet)
-    export_legend = str(export_legend)
+    export_legend_str = str(export_legend)
```
```suggestion
    export_legend = str(export_legend)
```
I have another suggestion below to fix the check below for `export_legend` (so that it checks for the str value) - so this variable can stay as-is so that it matches the formatting of the rest of the params.

lizgehret commented on gut_to_soil_manuscript_figures/_methods.py:
```diff
@@ -64,7 +63,7 @@ def pcoa_2d(output_dir: str, metadata: qiime2.Metadata,
         himalaya,
         pit_toilet,
-        export_legend,
+        export_legend_str,
```
```suggestion
        export_legend,
```

lizgehret commented on gut_to_soil_manuscript_figures/_methods.py:
```diff
@@ -44,12 +44,11 @@ def pcoa_2d(output_dir: str, metadata: qiime2.Metadata,
 
-    if export_legend:
-        legend_fp = os.path.join(output_dir, 'legend.png')
+    legend_fp = os.path.join(output_dir, 'legend.png')
```
```suggestion
```
This doesn't need to get assigned unless `export_legend` is set to True - so I'm going to move it down to the branch that also adds the legend as a separate item in the index.html page.

lizgehret commented on gut_to_soil_manuscript_figures/_methods.py:
```diff
@@ -80,6 +79,13 @@ def pcoa_2d(output_dir: str, metadata: qiime2.Metadata,
             <h1>2D PCoA Plot</h1>
             <img src="pcoa_plot.png" alt="PCoA Plot">
+        ''')
+        if export_legend:
```
```suggestion
        if export_legend == 'True':
            legend_fp = os.path.join(output_dir, 'legend.png')
```

lizgehret replied:
Now this functions essentially the same (after it's cast to a string) - this also matches the other checks I have in place in the actual script itself.

lizgehret commented on gut_to_soil_manuscript_figures/scripts/plot_pcoa_2d.py:
```diff
@@ -62,8 +63,7 @@ def _bucket_util(highlighted_buckets, md, ord_2d):
             md[(md['Bucket'] == bucket) &
-               (md['SampleType2'] == 'Self Sample') &
-               (md['Week'] == 0.0)].index.values
+               (md['SampleType'] == 'Human Excrement')].index.values
```
It looks like the `(md['Week'] == 0.0)].index.values` got removed. Was this intentional?

lizgehret commented on gut_to_soil_manuscript_figures/scripts/plot_pcoa_2d.py:
```diff
@@ -73,17 +73,16 @@ def _bucket_util(highlighted_buckets, md, ord_2d):
             md[(md['Bucket'] == bucket) &
-               (md['SampleType2'] == 'Bulking Material') &
-               (md['Week'] == 0.0)].index.values
+               (md['SampleType'] == 'Bulking Material')].index.values
```
Same comment as above, it looks like `(md['Week'] == 0.0)].index.values` was removed here as well.

lizgehret commented on gut_to_soil_manuscript_figures/scripts/plot_pcoa_2d.py:
```diff
@@ -233,8 +235,7 @@ def plot_pcoa_2d(metadata_fp, ordination_fp, measure,
             HE_week0 = \
-                md[(md['SampleType2'] == 'Self Sample') &
-                   (md['Week'] == 0.0)].index.values
+                md[(md['SampleType'] == 'Human Excrement')].index.values
```
Same comment re: `(md['Week'] == 0.0)].index.values`

lizgehret commented on gut_to_soil_manuscript_figures/scripts/plot_pcoa_2d.py:
```diff
@@ -245,8 +246,7 @@ def plot_pcoa_2d(metadata_fp, ordination_fp, measure,
             bulk_week0 = \
-                md[(md['SampleType2'] == 'Bulking Material') &
-                   (md['Week'] == 0.0)].index.values
+                md[(md['SampleType'] == 'Bulking Material')].index.values
```
Same comment re: `(md['Week'] == 0.0)].index.values`

lizgehret reviewed (changes requested):
i took a closer look at the `export_legend` stuff we were discussing @gregcaporaso - here are more detailed comments.

gregcaporaso committed:
Update gut_to_soil_manuscript_figures/_methods.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

gregcaporaso committed:
Update gut_to_soil_manuscript_figures/_methods.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

gregcaporaso committed:
Update gut_to_soil_manuscript_figures/_methods.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

gregcaporaso committed:
Update gut_to_soil_manuscript_figures/_methods.py

Co-authored-by: Liz Gehret <54517601+lizgehret@users.noreply.github.com>

gregcaporaso committed:
legend_fp was referenced before assignment

gregcaporaso committed:
address issue with missing Week 0.0 checks

gregcaporaso commented:
@lizgehret, thanks for catching the Week 0.0 issue! I think I fixed it now - we replaced those with empty cells, so checks for `week == 0.0` are now checks for `Composting Time Point is np.nan`. 

Here's the broken version I had yesterday for bucket 7:
![image](https://github.com/user-attachments/assets/a26c2b74-73b0-44e8-bb04-9e0abb6aaadd)

Here's today's fixed version:
![image](https://github.com/user-attachments/assets/dd88bfc0-75ab-4e0d-ad5e-a8c4ebd63f5b)

And here is one I'm pulling from Basecamp which I believe is the target:
<img width="808" alt="Screenshot 2024-10-09 at 7 14 33 AM" src="https://github.com/user-attachments/assets/39fb0219-6370-4114-8c7f-7945274fba74">

Note that in our metadata clean-up process, a few of the FLWC samples were removed so that is almost certainly the source of minor differences in the ordination.

lizgehret commented:
@gregcaporaso this does look better - the only thing I'm wondering about is the number of FLWC samples that are in your updated version (in comparison to the version from BC). It appears there's a higher number of that sample type than there was before - is that expected? We may want to do a quick tally of the total number we're expecting and visually see if that matches up with the updated plot.

gregcaporaso commented:
Replaced by #4
